sdk/monitor/monitor-query-metrics/samples/v1/javascript/queryWithAdvancedOptions.js

```diff
@@ -0,0 +1,159 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/**
+ * @summary Demonstrates advanced metrics querying with filtering, ordering, and rollup
+ */
+
+const { DefaultAzureCredential } = require("@azure/identity");
+const { MetricsClient, Durations } = require("@azure/monitor-query-metrics");
+require("dotenv/config");
+
+const resourceIds = process.env.METRICS_RESOURCE_IDS?.split(",") || [];
+const metricsNamespace = process.env.METRICS_RESOURCE_NAMESPACE || "";
+const endpoint = process.env.METRICS_ENDPOINT || "";
+
+async function main() {
+  if (resourceIds.length === 0) {
+    throw new Error(
+      "METRICS_RESOURCE_IDS must be set in the environment for this sample. " +
+        "Provide comma-separated resource IDs.",
+    );
+  }
+
+  if (!metricsNamespace) {
+    throw new Error("METRICS_RESOURCE_NAMESPACE must be set in the environment for this sample.");
+  }
+
+  if (!endpoint) {
+    throw new Error("METRICS_ENDPOINT must be set in the environment for this sample.");
+  }
+
+  const credential = new DefaultAzureCredential();
+  const metricsClient = new MetricsClient(endpoint, credential);
+
+  // Set up time range - last 24 hours
+  const endTime = new Date();
+  const startTime = new Date(endTime.getTime() - 24 * 60 * 60 * 1000); // 24 hours ago
+
+  console.log("Querying metrics with advanced parameters...");
+  console.log(`Time range: ${startTime.toISOString()} to ${endTime.toISOString()}`);
+  console.log(`Interval: ${Durations.oneHour} (1 hour)`);
+  console.log(`Resources: ${resourceIds.length} resources`);
+  console.log("Advanced options: Top 5 results, ordered by count descending");
+
+  try {
```

Comment:
can we get rid of all these tries?

Reply:
updates all samples-dev and published them into samples/v1